frontend/pages/dashboard/my-feed.tsx

```diff
@@ -3,28 +3,41 @@ import { NextPage } from 'next'
 import { withApollo } from '../../lib/apollo'
 import DashboardLayout from '../../components/Layouts/DashboardLayout'
 import MyFeed from '../../components/Dashboard/MyFeed'
-import { useFeedQuery } from '../../generated/graphql'
+import { useFeedQuery, useCurrentUserQuery } from '../../generated/graphql'
 import LoadingWrapper from '../../components/LoadingWrapper'
 import { useTranslation } from '../../config/i18n'
+import AuthGate from '../../components/AuthGate'
 
 interface InitialProps {
   namespacesRequired: string[]
 }
 
 const MyFeedPage: NextPage<InitialProps> = () => {
   const { t } = useTranslation()
-  const { loading, error, data } = useFeedQuery({
+  const {
+    data: currentUserData,
+    loading: currentUserLoading,
```

Comment:
Maybe I don't yet quite understand the structure, but I'm not sure why we should need to have a `useCurrentUserQuery` in any of the pages if `AuthGate` already makes that API call and handles loading/errors. I think what would happen in this file is that `useCurrentUserQuery` would be triggered, but `AuthGate` would immediately render and also make that call.

If we make `AuthGate` a render prop component, which exposes `currentUser` to the children, then the pages could immediately grab `currentUser` and not have to handle `loading` and `error` states in addition to any other `useXQuery`'s `loading` and `error` states. In the render prop case, a logged out user would just immediately get redirected, and for a logged in user, the cached value would be immediately available to the children.

```jsx
<AuthGate>
  {(currentUser) => (
    <DashboardLayout>
      <div>{currentUser.name} is the best}</div>
    </DashboardLayout>
  )
</AuthGate>
```

Let me know if that makes sense! 🙂

Reply:
Yeah, passing down currentUser makes sense to me 👍 . I don't think there's much of a perf difference since one one fetch should ever go through, but not having to handle loading/error states would be a win.

Reply:
Yeah @nlitwin as mentioned in the other PR that we closed earlier, we tested that no further calls are made with queries that have already been cached which navigating between pages in the app, so the additional calls would not be made. But yep I can work on simplifying it and passing down `currentUser`.

Reply:
Been having issues getting this all to work as requested as well, so unfortunately this is going to get held up another day now. Will get back to it tomorrow

Reply:
Sounds good 👍